### PR TITLE
Refactor docs

### PR DESCRIPTION
**Pull request by rpanackal**
## Context

AI/ai-sdk-java-backlog#68.

Improve documentation for readability.

### Feature scope:
 
- [x] Split content into multiple .md files by modules 
- [x] Add a quickstart and table of contents
- [x] Reduce redundancy eg requirements list
- [x] Move contribution instruction to Contribution Guide

## Definition of Done

- [x] Functionality scope stated & covered
- [x] Tests cover the scope above
- [x] ~Error handling created / updated & covered by the tests above~
- [x] ~Aligned changes with the JavaScript SDK~
- [x] ~Documentation updated~
- [x] ~Release notes updated~

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -5,641 +5,291 @@
...
+- [Code Snippets](#code-snippets)
+- [Contribute, Support and Feedback](#contribute-support-and-feedback)
+- [Security / Disclosure](#security--disclosure)
+- [Code of Conduct](#code-of-conduct)
+- [Licensing](#licensing)
+
+## Introduction
+
+Welcome to the **SAP Cloud SDK for AI (for Java)**. This SDK enables developers to seamlessly integrate AI capabilities,
+such as chat completion, into their Java-based business applications using SAP's Generative AI Hub. Leverage powerful
```
Comment (newtork) [inline README.md:34]: I think we can't use the term "Generative AI Hub".

File: README.md [hunk 2]
```diff
@@ -5,641 +5,291 @@
...
-        .orchestrationV1EndpointsCreate(config);
-
-final String messageResult =
-    result.getOrchestrationResult().getChoices().get(0).getMessage().getContent();
+import com.sap.ai.sdk.foundationmodels.openai.OpenAiClient;
+import com.sap.ai.sdk.foundationmodels.openai.OpenAiModel;
+import com.sap.ai.sdk.foundationmodels.openai.model.OpenAiChatCompletionOutput;
+
+public class OpenAiExample {
+  public static void main(String[] args) {
```
Comment (newtork) [inline README.md:199]: (Minor) 

Please remove imports and psvm (`public static void main`) boilerplate. The code is easy enough, I think.

File: README.md [hunk 3]
```diff
@@ -5,641 +5,291 @@
...
-```
+### Prerequisites
 
-See [an example pom in our Spring Boot application](sample-code/spring-app/pom.xml)
+Before you begin, ensure you have:
 
-### Chat completion template
+- Met the [General Requirements](#general-requirements).
+- Set up the AI Core credentials
+  using [Option 1: Set Credentials as Environment Variable](#option-1-set-credentials-as-environment-variable).
```
Comment (newtork) [inline README.md:182]: (Minor)

```suggestion
  using [(1) Environment Variable](#option-1-set-credentials-as-environment-variable) or [(2) Regular Service Binding](#option-1-set-credentials-as-environment-variable).
```

I would consider the second option also low effort and convenience enough to mention it here similarly to first option.

File: README.md [hunk 4]
```diff
@@ -5,641 +5,291 @@
...
+```shell
+cd sample-code/spring-app
+mvn spring-boot:run
 ```
 
-See [an example in our Spring Boot application](sample-code/spring-app/src/main/java/com/sap/ai/sdk/app/controllers/OrchestrationController.java)
+### Deploying to Cloud Foundry (Optional)
 
-### Data masking
+When deploying your application to Cloud Foundry, use **service binding** to provide the AI Core credentials as per
```
Comment (newtork) [inline README.md:228]: (Minor)

```suggestion
When deploying your productive application to Cloud Foundry, it's recommended to use **service binding** to provide the AI Core credentials as per
```

File: README.md [hunk 5]
```diff
@@ -5,641 +5,291 @@
...
-                "temperature", 0.1,
-                "frequency_penalty", 0,
-                "presence_penalty", 0));
-```
+- [OpenAI Chat Completion](docs/guides/OPENAI_CHAT_COMPLETION.md)
+- [Orchestration Chat Completion](docs/guides/ORCHESTRATION_CHAT_COMPLETION.md)
+- [AI Core Deployment](docs/guides/AI_CORE_DEPLOYMENT.md)
 
-See [an example in our unit test](orchestration/src/test/java/com/sap/ai/sdk/orchestration/client/OrchestrationUnitTest.java)
+## Code Snippets
```
Comment (newtork) [inline README.md:251]: (Minor/Comment)

I would maybe rephrase this section as FAQ for now. (Even when the questions haven't been asked frequently.)

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -37,6 +37,11 @@ The following rule governs code contributions:
 * Contributions must be licensed under the [Apache 2.0 License](./LICENSE)
 * Due to legal reasons, contributors will be asked to accept a Developer Certificate of Origin (DCO) when they create the first pull request to this project. This happens in an automated fashion during the submission process. SAP uses [the standard DCO text of the Linux Foundation](https://developercertificate.org/).
 
+### Set-up Formatting
+
+- Install the Google Java Format plugin on Intellij and follow
```
Comment (newtork) [inline CONTRIBUTING.md:42]:
```suggestion
It's recommended to install the [Google Java Format plugin for IntelliJ](https://plugins.jetbrains.com/plugin/8527-google-java-format), while following
```

File: README.md (hunk 5 above)
Comment (rpanackal) [reply README.md:251]: How about 'Troubleshooting'?

File: docs/guides/AI_CORE_DEPLOYMENT.md [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+See [an example pom in our Spring Boot application](../../sample-code/spring-app/pom.xml)
+
+## Usage
+
+## Create a Deployment
+
+Use the following code snippet to create a deployment in SAP AI Core:
+
+```java
+public AiDeploymentCreationResponse createDeployment() {
```
Comment (newtork) [inline docs/guides/AI_CORE_DEPLOYMENT.md:74]: (Minor)

I understand you copied it from the previous `README.md` but at this point I would like to ask to drop the method boilerplate.

And to extract a variable for `api = new DeploymentApi(getClient())`, this will slightly reduce the complexity.

And to extract a variable for `resourceGroup = "default"`, this will reduce ambiguousness.

File: docs/guides/AI_CORE_DEPLOYMENT.md [hunk 8]
```diff
@@ -4,22 +4,40 @@
...
+the [README.md](../../README.md#general-requirements). Additionally, include the
+necessary Maven dependency in your project.
 
-Additionally, to carry out the examples in this guide, you need:
+### Maven Dependencies
+
+Add the following dependency to your `pom.xml` file:
+
+```xml
+
```
Comment (newtork) [inline docs/guides/AI_CORE_DEPLOYMENT.md:27]:
```suggestion
```

File: README.md [hunk 9]
```diff
@@ -5,641 +5,291 @@
...
-final CompletionPostResponse result =
-    new OrchestrationCompletionApi(getOrchestrationClient("default"))
-        // this fails with Bad Request because the strict filter prohibits the input message
-        .orchestrationV1EndpointsCreate(config);
-
-final String messageResult =
-    result.getOrchestrationResult().getChoices().get(0).getMessage().getContent();
+```shell
+cd sample-code/spring-app
+mvn spring-boot:run
```
[the remark this answers is not in the document]
Comment (rpanackal) [reply README.md:209]: Yes. `mvn clean spring-boot:run` works

**Review by newtork — APPROVED** [review]
